pkg/gateway/gateway_test.go

```diff
@@ -4,11 +4,238 @@ package gateway
 // Licensed under the Apache License 2.0.
 
 import (
+	"context"
 	"testing"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/golang/mock/gomock"
+	"github.com/pires/go-proxyproto"
+	"github.com/sirupsen/logrus"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
+	mock_database "github.com/Azure/ARO-RP/pkg/util/mocks/database"
+	mock_env "github.com/Azure/ARO-RP/pkg/util/mocks/env"
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+	utilnet "github.com/Azure/ARO-RP/pkg/util/net"
 )
 
+func TestNewGatway(t *testing.T) {
```

Comment:
The smallest nit in the world. Needs an "e". Otherwise lgtm and thanks for adding all these tests.